### PR TITLE
Popover Functionality

### PR DESCRIPTION
**Pull request by feydkin**
Looking for feedback / direction on this implementation of a popover feature.

### CODE REVIEW

**Comment by undsoft** [comment]
How about just installing https://github.com/MrFrankel/ngx-popper?
I've had good experience with it in the past. It's a port of https://popper.js.org.
We can adjust styles to match our website.

**Comment by feydkin** [comment]
@undsoft - I've installed ngx-popper best I know how. It works. Any suggestions beyond styling? 
ngx-popper

File: src/assets/css/_popper.scss [hunk 1]
```diff
@@ -0,0 +1,12 @@
+@import "variables";
+
+.infoIcon {
+  padding-left: 10px;
+  display: inline-block;
+}
+
+.popperStyle {
```
Comment (undsoft) [inline src/assets/css/_popper.scss:8]: Class names are usually snake-case.

File: src/app/pages/stats-page/stats-page.component.html [hunk 2]
```diff
@@ -31,15 +41,46 @@
       <div class="break"></div>
       <div class="big-number">
         <div class="big-number-value">{{ info.minutes_to_win | minutes }}</div>
-        <div class="big-number-title">Estimated Time To Win</div>
+        <div class="big-number-title">
+          Estimated Time To Win
+          <div class="infoIcon" [popper]="tooltipTimeToWin"
+               [popperTrigger]="'hover'"
```
Comment (undsoft) [inline src/app/pages/stats-page/stats-page.component.html:47]: 1. You can just do `popperTrigger="hover"` in such cases.
2. You can define default settings, such as trigger, etc in `NgxPopperModule.forRoot({})`.

File: src/app/pages/stats-page/stats-page.component.html [hunk 3]
```diff
@@ -31,15 +41,46 @@
...
+               [popperPlacement]="'bottom'"
+               [popperApplyClass]="'popperStyle'">
+               <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" viewBox="0 0 16 16">
+                <path d="M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412-1 4.705c-.07.34.029.533.304.533.194 0 .487-.07.686-.246l-.088.416c-.287.346-.92.598-1.465.598-.703 0-1.002-.422-.808-1.319l.738-3.468c.064-.293.006-.399-.287-.47l-.451-.081.082-.381 2.29-.287zM8 5.5a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"/>
```
Comment (undsoft) [inline src/app/pages/stats-page/stats-page.component.html:51]: We've been dancing around icons for a while. I'd be okay with just installing material icons.
If this is not something you want to do, try to find a way to not repeat this in a template, for example by extracting to the component.

**Review by undsoft — COMMENTED** [review]
Looks good so far.

File: src/app/pages/stats-page/stats-page.component.html (hunk 3 above)
Comment (feydkin) [reply src/app/pages/stats-page/stats-page.component.html:51]: Installed material icons and extracted to a template. I'm still messing around with styles so I will be a couple more days on this.

**Comment by feydkin** [comment]
- Added material icons
- Fixed up popper implementation, **need help with approach to styling**

File: src/app/components/poptip/poptip.component.html [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<popper-content #poptip>
+    <p>{{msg}}</p>
+</popper-content>
+<div class="info-icon"  [popper]="poptip"
+                        [popperTrigger]="'hover'"
+                        [popperHideOnClickOutside]="true"
+                        [popperHideOnScroll]="true"
+                        [popperPlacement]="'right'"
+                        [popperApplyClass]="'test-class'">
```
Comment (undsoft) [inline src/app/components/poptip/poptip.component.html:9]: Just popperPlacement="right"

**Review by undsoft — COMMENTED** [review]
If the question about styling is how it should look, then I'd prefer smaller font, fewer padding and a more comparable size between tooltip arrow and rest of the box.
If the question is how to implement it technically, then it would be appropriate to create a global cxss file in `assets/css/elements` and add styles there.

File: src/app/components/poptip/poptip.component.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<popper-content #poptip>
+    <p>{{msg}}</p>
+</popper-content>
+<div class="info-icon"  [popper]="poptip"
+                        [popperTrigger]="'hover'"
```
Comment (undsoft) [inline src/app/components/poptip/poptip.component.html:5]: Keep in mind that some settings can be set as system-wide defaults in `NgxPopperModule.forRoot({})`

File: src/app/components/poptip/poptip.component.scss [hunk 6]
```diff
@@ -0,0 +1,4 @@
+.info-icon {
+  padding-left: 10px;
+  display: inline-block;
+}
```
Comment (undsoft) [inline src/app/components/poptip/poptip.component.scss:4]: Need different cursor here and need to align it with text.

File: src/app/pages/stats-page/stats-page.component.html [hunk 7]
```diff
@@ -13,7 +13,10 @@
       </div>
       <div class="big-number">
         <div class="big-number-value">{{ info.blockchain_totalspace | filesize: { standard: 'iec' } }}</div>
-        <div class="big-number-title">Total Network Space</div>
+        <div class="big-number-title">
+          Total Network Space
+          <poptip msg="The combine estimated size of all storage used on the Chia Network."></poptip>
```
Comment (undsoft) [inline src/app/pages/stats-page/stats-page.component.html:18]: The combined

File: src/app/components/poptip/poptip.component.html [hunk 8]
```diff
@@ -0,0 +1,10 @@
+<popper-content #poptip>
+    <p>{{msg}}</p>
+</popper-content>
+<div    class="info-icon material-icons-outlined"
```
Comment (undsoft) [inline src/app/components/poptip/poptip.component.html:4]: 2 spaces please

**Review by undsoft — APPROVED** [review]
Looks good!

**Comment by kmoore134** [comment]
@feydkin - If you want to resole the conflicts, I can get this merged.

**Comment by feydkin** [comment]
> 
> 
> @feydkin - If you want to resolve the conflicts, I can get this merged.

Done. Thank you.

**Pull request merged**